### PR TITLE
Fix broken PropTypes (regression from #4408)

### PR DESCRIPTION
**Pull request by ykzts**

### CODE REVIEW

**Comment by ykzts** [comment]
cc @sorin-davidoi

File: app/javascript/mastodon/components/status.js [hunk 1]
```diff
@@ -41,8 +41,8 @@ export default class Status extends ImmutablePureComponent {
     autoPlayGif: PropTypes.bool,
     muted: PropTypes.bool,
     intersectionObserverWrapper: PropTypes.object,
-    index: PropTypes.oneOf(PropTypes.string, PropTypes.number),
-    listLength: PropTypes.oneOf(PropTypes.string, PropTypes.number),
+    index: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
```
Comment (ykzts) [inline app/javascript/mastodon/components/status.js]: ref https://facebook.github.io/react/docs/typechecking-with-proptypes.html

**Comment by unarist** [comment]
Actually, should it be "string or number" instead of "number"?

**Comment by sorin-davidoi** [comment]
We are not doing any computations with it, and it gets converted to a string anyway (HTML attribute). Wanted to avoid manually casting it like `<Status index={String(i)} />`.

**Review by unarist — APPROVED** [review]

**Pull request merged**